src/pybind/mgr/volumes/fs/async_job.py

```diff
@@ -63,6 +68,10 @@ def run(self):
         log.error("thread [{0}] reached exception limit, bailing out...".format(thread_name))
```

Comment:
Should we add a call to `self.async_job.threads.remove(self)` here, as the thread is bailing out? Future increase/decrease of threads will not account for self terminated threads if not taken out of the list.

Reply:
Good catch. It was there in some iteration of the patch. Somehow got missed. I will fix it.

Reply:
Oh this is on error condition. I misunderstood for reconfigure. I will add it.